lib/modules/datasource/go/releases-goproxy.ts

```diff
@@ -172,7 +172,9 @@ export class GoProxyDatasource extends Datasource {
   static parsedNoproxy: Record<string, RegExp | null> = {};
 
   static parseNoproxy(
-    input: unknown = process.env.GONOPROXY || process.env.GOPRIVATE
+    input: unknown = process.env.GONOPROXY ||
+      process.env.GOPRIVATE ||
+      process.env.GOINSECURE
```

Comment:
We need a Go expert to confirm if this logic is correct

Reply:
Docs say:

> GOINSECURE is a new environment variable that instructs the go command to not require an HTTPS connection, and to skip certificate validation, when fetching certain modules directly from their origins.

This seems unrelated to GONOPROXY and GOPRIVATE, and therefore the above change will product incorrect behavior

Reply:
yes, you are right,  `GONOPROXY`, `GOPRIVATE` and `GONOSUMDB` are related but `GOINSECURE` is not related to the previous three env variables.